src/redux/nationSlice/NationSlice.jsx

```diff
@@ -0,0 +1,87 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const API_NOBEL = 'https://api.nobelprize.org/2.1/laureates';
+
+const initialState = {
+  nobels: [],
+  isLoading: true,
+  amount: 0,
+  uniqueCountries: [],
+};
+
+export const fetchNobel = createAsyncThunk(
+  'nobelSlice/fetchNobel',
+  async (name, thunkAPI) => {
+    try {
+      const res = await axios.get(API_NOBEL);
+      return res.data;
+    } catch (err) {
+      return thunkAPI.rejectWithValue('st went wrong');
+    }
+  },
+);
+
+export const takeUniqueCountries = (arr) => {
+  const uniqueCountries = [];
+  const compare = [];
+  // console.log('takeUniqueCountries'.toUpperCase());
+
+  arr.forEach(({ country, id, city }) => {
+    if (!compare.includes(country)) {
+      compare.push(country);
+      uniqueCountries.push({ id, country, cities: [city] });
+    } else {
+      const check = uniqueCountries.find((item) => item.country === country);
+      check.cities.push(city);
+    }
+  });
+  return uniqueCountries;
+};
+
+const nobelSlice = createSlice({
+  name: 'nobelSlice',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchNobel.pending, (state) => ({
+        ...state,
+      }))
+      .addCase(fetchNobel.fulfilled, (state, { payload }) => {
+        const arr = payload.laureates.map((item) => {
+          const { id, fullName: { en: fullname }, gender } = item;
+          const { date } = item.birth;
+          const { en: city } = item.birth.place?.cityNow || { en: 'Unknown' };
+          const { en: country } = item.birth.place?.countryNow || { en: 'Unknown' };
+
+          const {
+            awardYear,
+            categoryFullName: { en: category },
+            prizeAmount: prize,
+          } = item.nobelPrizes[0];
+
+          // const city = city123.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+          // const country = country123.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+
```

Comment:
- [ ] It s against the best practices to leave unused code as a comment on your code base. I'll suggest you remove this as it makes the codebase look a bit unprofessional. :+1: :smile: :wave: